pkg/neonvm/cpuscaling/sysfs.go

```diff
@@ -0,0 +1,195 @@
+package cpuscaling
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// CPU directory path
+const cpuPath = "/sys/devices/system/cpu/"
+
+type CPUSysFsStateScaler struct{}
+
+func (c *CPUSysFsStateScaler) EnsureOnlineCPUs(targetCount int) error {
+	cpus, err := getAllCPUs()
+	if err != nil {
+		return err
+	}
+
+	onlineCount, err := c.ActiveCPUsCount()
+	if err != nil {
+		return err
+	}
+
+	if onlineCount < uint32(targetCount) {
+		for _, cpu := range cpus {
+			if cpu == 0 {
```

Comment:
suggestion: this is a lot of nested control flow. Maybe worth separating into >1 function?

*(recreated from https://github.com/neondatabase/autoscaling/pull/1111#discussion_r1804084387)*

Reply:
Changed the code in this layer:
- split into two structures, one that reconciles the amount of CPUs and the other one works with sysfs files
- added unit tests for the reconcilation logic
- changed functions to work with ranges instead of slices.